host/src/protocol/handler.cpp

```
@@ -0,0 +1,173 @@
+private:
+    static bool process_result(core::protocol::Serializer::SerializeResult result) {
+        using core::protocol::Serializer;
+        if (result == Serializer::SerializeResult::kSuccess) [[likely]]
+            return true;
+        if (result == Serializer::SerializeResult::kBadAlloc) {
+            logging::get_logger().error("Transmit buffer unavailable (acquire failed)");
+            return true;
+        }
+        if (result == Serializer::SerializeResult::kInvalidArgument) {
+            return false;
+        }
```

Cppcheck (2.19.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)